File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(const std::vector<int>& prices) {
    if (prices.empty()) {
        return 0;  // No prices available, so no profit can be made
    }

    int minPrice = prices[0];
    int maxProfit = 0;

    for (int i = 1; i < prices.size(); i++) {
        int currentPrice = prices[i];

        if (currentPrice < minPrice) {
            minPrice = currentPrice;  // Update the minimum price
        } else {
            int profit = currentPrice - minPrice;
            maxProfit = std::max(maxProfit, profit);  // Update the maximum profit
        }
    }

    return maxProfit;
}
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) ==0:
            return 0

        minprice = prices[0]
        maxprofit = 0

        for i in range(1,len(prices)):
            if prices[i] < minprice:
                minprice = prices[i]
            else:
                profit = prices[i] - minprice
                maxprofit = max(maxprofit, profit)

        return maxprofit